### ISSUE TITLE
multilingual products

### ISSUE THREAD

**Issue by Remo** [VividWeb/vivid_store#74]
Have you thought about adding multi lingual product descriptions? It might be possible to use something like this https://github.com/mlocati/concrete5-multilingual_attributes but I think it would be benefical to have true multilingual support in the store meaning pages get published underneath /en /de, make option groups transtable and so on..

**Comment by brucewyne** [comment]
![Heavy](http://www.jaredmcdaris.com/wp-content/uploads/2014/08/h36EF146D.jpeg)

**Comment by Remo** [comment]
yes sir (: But it's pretty essential for us. I'm aware that being from Switzerland, a country with a population of 8mio and 4 official languages, makes me a bit different regarding this need, but this is pretty much a deal breaker for us.

**Comment by brucewyne** [comment]
I imagine you're going to bring a few international ideas that I've just not considered much before. Which is a great thing, just complex in nature in handling right. 

What do you typically use now for ecommerce? AFAIK old 5.6 ecom didn't do this, but nor do many typical ecom packages that I can think of off the top of my head. While I understand the end result, I'd love to see some other examples of implementation.

Thanks for you feedback in general. Helps us know what we don't know.

**Comment by Remo** [comment]
A screenshot from our store:
![image](https://cloud.githubusercontent.com/assets/129864/8006243/4b770f04-0b92-11e5-9c0f-97d521fee36f.png)

We also have a screen where you can select attributes which are identical in all languages. When you save a product in a certain language, those get transferred to all other languages too:
![image](https://cloud.githubusercontent.com/assets/129864/8006263/81e496c4-0b92-11e5-9615-be063dcc60a5.png)

**Comment by Remo** [comment]
Most store, even bigger solutions like shopify and ecwid don't have proper support for multi lingual products which is often a critical issue for us. One of the main reasons why we've ended up creating our own store for 5.6..

**Comment by brucewyne** [comment]
Do you recreate everything for the additional language? Or is there some stuff more global than others? E.g. Descriptions/attributes are translated and two different forms, quantities/images aren't..

**Comment by Remo** [comment]
Behind those product languages are concrete5 pages. One for each language. The data is pretty redundant, but controlled through the UI. Product categories (groups in your case) as well as products were all pages in our system. A product assigned to categories was an alias to get a proper page structure and make autonav usable.

**Comment by Remo** [comment]
# Multi lingual attributes
- This would be easy as we could use the existing product classes
- It would require us to create a number of attributes
- Using existing attributes would be hard
- The output would have to be customized, bascially check multilingual section and if one exists try to get the right value from an each attribute

# A product per language
- Creating a DB object for each product language gives us a lot of flexibility
- There's a bit of overhead to connect the products into a single entity
- Editing in the front end works nicely as we can use the multilingual component to switch between languages
- Also creates a bit of redundancy
- Attributes which aren't language specific would have to be synced
- We could easily create pages for product groups and add aliases for products beneath them. That would make it possible to use autonav

**Comment by brucewyne** [comment]
Just wanted to comment and say I've been thinking about this a lot lately and how to approach it in a nice way that's not so redundant. I also have to admit I don't deal with multilingual sites. I haven't done one in at least 5 years. So my knowledge with it especially in C5 is not intimate. I'm going to spend some time exploring C5 as it is, how it handles multilingual sites, and what approach would make sense for this.

**Comment by brucewyne** [comment]
Circling back to this. One of the biggest problems that I can't wrap my head around is the single pages such as the modal, cart & checkout. Do you think that could be solved with a query string in the url rather than a separate page?

e.g. http://domain.com/checkout?lang=EN_US

**Comment by Remo** [comment]
This shouldn't be necessary, the last language from a multilingual section should be used for routes that aren't part of a mapped multilingual section. Adding a language parameter was what I did while concrete5 still had a ton of multilingual related bugs.
